models/baseModels/Lead/Lead.ts

```diff
@@ -0,0 +1,116 @@
+import { Fyo } from 'fyo';
+import { Doc } from 'fyo/model/doc';
+import {
+  Action,
+  LeadStatus,
+  ListViewSettings,
+  ReadOnlyMap,
+  ValidationMap,
+} from 'fyo/model/types';
+import { getLeadStatusColumn } from 'models/helpers';
+import {
+  validateEmail,
+  validatePhoneNumber,
+} from 'fyo/model/validationFunction';
+import { DocValueMap } from 'fyo/core/types';
+import { ModelNameEnum } from 'models/types';
+
+export class Lead extends Doc {
+  status?: LeadStatus;
+
+  getCustomerDoc(): Lead | undefined {
+    if (!this.name) return;
+
+    const docData = this.getValidDict(true, true);
+
+    if (!docData) {
+      return;
+    }
+    const partyDocData = {
+      ...docData,
+      role: 'Customer',
+      fromLead: this.name,
+    } as DocValueMap;
+    const newPartyDoc = this.fyo.doc.getNewDoc(
+      ModelNameEnum.Party,
+      partyDocData
+    ) as Lead;
+    return newPartyDoc;
+  }
+
+  getSalesQuoteDoc(itemData: { name: string; rate: number }): Lead | undefined {
```

Comment:
why mention `itemData` here?.  Ideally `getSalesQuoteDoc` shouldn't be receiving any props.